Source/BaseTableDataDisplayManager/BaseTableDataDisplayManager.swift

```diff
@@ -323,6 +324,38 @@ extension BaseTableDataDisplayManager: UITableViewDelegate {
         self.scrollEvent.invoke(with: guardTable)
     }
 
+    open func tableView(_ tableView: UITableView, canMoveRowAt indexPath: IndexPath) -> Bool {
+        if let generator = cellGenerators[indexPath.section][indexPath.row] as? MovableGenerator {
+            return generator.canMove()
+        }
+        return false
+    }
+
+    open func tableView(_ tableView: UITableView, moveRowAt sourceIndexPath: IndexPath, to destinationIndexPath: IndexPath) {
+        guard sourceIndexPath.section == destinationIndexPath.section else {
+            return
+        }
+        let section = sourceIndexPath.section
+        var oldCellGenerators = cellGenerators[section]
+        let itemToMove = oldCellGenerators[sourceIndexPath.row]
+        oldCellGenerators.remove(at: sourceIndexPath.row)
+        oldCellGenerators.insert(itemToMove, at: destinationIndexPath.row)
+
+        let sectionGenerator = sectionHeaderGenerators[section]
+        removeAllGenerators(in: sectionGenerator)
+        addCellGenerators(oldCellGenerators, toHeader: sectionGenerator)
+
+        cellChangedPosition.invoke(with: (section: section, oldIndex: sourceIndexPath.row, newIndex: destinationIndexPath.row))
+    }
+
+    open func tableView(_ tableView: UITableView, shouldIndentWhileEditingRowAt indexPath: IndexPath) -> Bool {
+        return false
+    }
+
+    open func tableView(_ tableView: UITableView, editingStyleForRowAt indexPath: IndexPath) -> UITableViewCell.EditingStyle {
+        return .none
```

Comment:
May we should configure style? 
If you want to move cell (with drag&drop) you should change editing style
May use `MovableGenerator` in this place?

Reply:
yes, that is good idea, but editing it is not about `MovableGenerator`. Maybe we should add `EditableGenerator` protocol to try to cast and get appropriate value?

Reply:
Mmm, but in this case Movable require Editable :)
May be needs to add conformance between `MovableGenerator` and `EditableGenerator`?

Reply:
ok, I will add this